src/main/java/com/apigee/edge/config/mavenplugin/CustomRoleMojo.java

```diff
@@ -0,0 +1,352 @@
+/**
+ * Copyright (C) 2016 Apigee Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apigee.edge.config.mavenplugin;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.apigee.edge.config.rest.RestUtil;
+import com.apigee.edge.config.utils.ServerProfile;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.util.Key;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+
+/**
+ * Goal to create Custom Roles in Apigee EDGE. scope: org
+ *
+ * @author Abhishek Chouksey
+ * @author Pallavi Tanpure
+ * @author Soudnya Nalawade
+ * @author Himanshu Sachdev
+ * @goal customroles
+ * @phase install
+ */
+
+public class CustomRoleMojo extends GatewayAbstractMojo {
```

Comment:
can we use UserRoleMojo in stead of CustomRoleMojo ?
And all the references in the code to use UserRoles in stead of Custom Roles